File: benchmark/src/nodejs/number_parse.js
const n=200000;let seed=12345;function ri(m){seed=(seed*16807)%2147483647;return seed%m}
const strings=[];for(let i=0;i<n;i++){let s="";if(ri(2)===0)s="-";let nd=1+ri(9);s+=String(1+ri(9));for(let j=1;j<nd;j++)s+=String(ri(10));strings.push(s)}
const start=Date.now();let total=0;const M=1000003
for(let si=0;si<n;si++){let s=strings[si],neg=false,idx=0;if(s[0]==='-'){neg=true;idx=1}let val=0
while(idx<s.length){let c=s[idx],d;if(c==='0')d=0;else if(c==='1')d=1;else if(c==='2')d=2;else if(c==='3')d=3;else if(c==='4')d=4;else if(c==='5')d=5;else if(c==='6')d=6;else if(c==='7')d=7;else if(c==='8')d=8;else d=9
val=val*10+d;idx++}if(neg)val=-val;total+=val}
let r=((total%M)+M)%M;const end=Date.now();console.log(`Result: ${r}`);console.log(`Time: ${end-start}ms`)

File: benchmarks/src/ruby/number_parse.rb
n = 200000
$seed = 12345
def ri(m)
  $seed = ($seed * 16807) % 2147483647
  $seed % m
end

strings = []
n.times do
  s = ""
  s = "-" if ri(2) == 0
  nd = 1 + ri(9)
  s << (1 + ri(9)).to_s
  (nd - 1).times { s << ri(10).to_s }
  strings << s
end

start = Time.now.to_f * 1000
total = 0
M = 1000003

strings.each do |s|
  neg = false
  idx = 0
  if s[0] == "-"
    neg = true
    idx = 1
  end
  val = 0
  while idx < s.length
    c = s[idx]
    d = case c
        when "0" then 0
        when "1" then 1
        when "2" then 2
        when "3" then 3
        when "4" then 4
        when "5" then 5
        when "6" then 6
        when "7" then 7
        when "8" then 8
        else 9
        end
    val = val * 10 + d
    idx += 1
  end
  val = -val if neg
  total += val
end

r = ((total % M) + M) % M
finish = Time.now.to_f * 1000
puts "Result: #{r}"
puts "Time: #{(finish - start).round}ms"
